packages/react/src/BlockSideMenu/components/DefaultButtons/BlockColorsButton.tsx

```diff
@@ -0,0 +1,80 @@
+import { useCallback, useRef, useState } from "react";
+import { Box, Menu } from "@mantine/core";
+import { HiChevronRight } from "react-icons/hi";
+import { ColorPicker } from "../../../SharedComponents/ColorPicker/components/ColorPicker";
+import {
+  DragHandleMenuItem,
+  DragHandleMenuItemProps,
+} from "../DragHandleMenuItem";
+
+export const BlockColorsButton = (props: DragHandleMenuItemProps) => {
+  const [block] = useState(props.editor.getMouseCursorPosition()?.block);
```

Comment:
why useState?

Reply:
It's because the menu has to update whenever we hover on/off the colors button, so without `useState`, `block` would get updated to whichever block the cursor is next to and the color might get applied to the wrong one.